lib/src/index.dart

```diff
@@ -414,33 +414,69 @@ class MeiliSearchIndex {
     return await _getTask(http.deleteMethod('/indexes/$uid/documents/$id'));
   }
 
-  /// Delete a selection of documents by given [ids].
-  Future<Task> deleteDocuments(List<Object> ids) async {
-    return await _getTask(
-      http.postMethod(
+  /// Delete a selection of documents by given query.
+  Future<Task> deleteDocuments([DeleteDocumentsQuery? query]) async {
+    Object? deleteBatchData;
+    Object? deleteData;
+    if (query != null) {
+      if (query.ids != null) {
+        // if ids are available, fallback to the delete-batch route
+        deleteBatchData = query.ids;
+      } else {
+        // if ids are NOT available, use the new delete route
+        deleteData = query.toSparseMap();
+      }
+    }
+
+    Future<Response<Map<String, Object?>>> future;
+    if (deleteData != null) {
+      future = http.postMethod(
+        '/indexes/$uid/documents/delete',
+        data: deleteData,
+      );
+    } else {
+      future = http.postMethod(
         '/indexes/$uid/documents/delete-batch',
-        data: ids,
-      ),
-    );
+        data: deleteBatchData,
+      );
+    }
+
+    return await _getTask(future);
```

Comment:
What do you think?

```
Future<Task> deleteDocuments([DeleteDocumentsQuery? query]) async {
  String route = 'delete';
  Object? data;

  if (query != null) {
    if (query.ids != null) {
      // if ids are available, fallback to the delete-batch route
      data = query.ids;
      route = '$route-batch';
    } else {
      // if ids are NOT available, use the new delete route
      data = query.toSparseMap();
    }
  }

  return await _getTask(
    http.postMethod('/indexes/$uid/documents/$route', data: data)
  );
}
```

Reply:
I updated the logic, and also added client-side assertions to prevent bad inputs, since the server will throw on them anyway

Reply:
relevant commit: https://github.com/meilisearch/meilisearch-dart/pull/318/commits/47eed42414d6d91cd32760af6f594e5a6e0271b5